File: Shared/Boards/Board.go
package boards

type BoardType string

const (
	Type_Drone    BoardType = "Drone"
	Type_Control  BoardType = "Control"
	Type_LandNode BoardType = "CentralBoard"
)

// Board 结构体用于表示装载在飞控上的可编程板, 飞控板本身,
type Board struct {
	BoardID   string    `json:"board_id"`
	BoardType BoardType `json:"board_type"`
	BoardName string    `json:"board_name"`
	BoardDesc string    `json:"board_desc"`

	BoardStatus string `json:"board_status"`
	BoardConfig string `json:"board_config"`
	BoardIP     string `json:"board_ip"`
	BoardPort   string `json:"board_port"`

	IsOnline    bool `json:"is_online"`
	IsConnected bool `json:"is_connected"`
}

File: src/types/board.ts
import type { ApiResponse, JsonObject } from './api'

export type BoardConnectionType = 'TCP' | 'UDP' | (string & {})
export type BoardType = 'Drone' | 'Sensor' | (string & {})

export interface CreateBoardParams {
  board_id: string
  board_name?: string
  board_type: BoardType
  connection: BoardConnectionType
  address: string
  port: string
}

export interface BoardInfo extends JsonObject {
  boardId: string
  boardName?: string
  boardType?: BoardType
  boardIp?: string
  boardPort?: string
  boardStatus?: string
  isConnected: boolean
}

export interface SendMessageParams {
  to_id: string
  to_type: string
  command: string
  attribute: string
  data: JsonObject
}

export type BoardListResponse = ApiResponse<{ boards: BoardInfo[] }>
export type BoardInfoResponse = ApiResponse<{ board: BoardInfo }>
export type CreateBoardResponse = ApiResponse<{ board_id: string; message?: string }>
export type SendMessageResponse = ApiResponse<{ message: string }>
export type StopBoardResponse = ApiResponse<{ message: string }>